x-hooks/use-debounced-callback.ts

```diff
@@ -0,0 +1,15 @@
+import { useRef, useCallback } from "react";
+
+export default function useDebouncedCallback(callback: (...args) => void, time: number) {
+  const timeout = useRef(null);
+
+  return useCallback((...args) => {
```

Comment:
we need a clearTimeout for the unmount, so we don't leave things hanging

That said, I think we can simply use the one from `use-debounce`